### PR TITLE
🥏 [C01] Slippage check on reward token swaps

### PR DESCRIPTION
**Pull request by tomlinton**
Add a swap token list to the vault and separate out the harvest/swap functions. The swap function uses chainlink oracles to determine the min expected output from Uniswap.

Also changed the approvals to use infinite approvals for vault -> uniswap when the swap token is added, and removes the approval when a swap token is removed. Previously it was doing approvals of the actual required amount every time it swapped. We save gas with the infinite approval but it is less secure, but I think it is OK since the swap tokens should never be sitting in the vault contract. Happy to discuss.

Contract change checklist:
  - [ ] Code reviewed by 2 reviewers. See [template](https://github.com/OriginProtocol/security/blob/master/templates/Contract-Code-Review.md) and [example](https://github.com/OriginProtocol/security/blob/master/templates/Contract-Code-Review-Example.md).
  - [ ] Unit tests pass
  - [ ] Slither tests pass with no warning
  - [ ] Echidna tests pass if PR includes changes to OUSD contract (not automated, run manually on local)

### CODE REVIEW

**Comment by tomlinton** [comment]
@DanielVF put this up for some early feedback. Are we on the same page with this? 😁

**Comment by DanielVF** [comment]
@tomlinton Much Good

**Review by DanielVF — APPROVED** [review]

File: contracts/contracts/vault/VaultAdmin.sol [hunk 1]
```diff
@@ -185,18 +193,86 @@ contract VaultAdmin is VaultStorage {
...
+
+    /**
+     * @dev Remove a swap token from the tokens that get liquidated for stablecoins.
+     * @param _addr Address of the token
+     */
+    function removeSwapToken(address _addr) external onlyGovernor {
+        bool added = false;
+        for (uint256 i = 0; i < swapTokens.length; i++) {
+            if (swapTokens[i] == _addr) {
+                added = true;
```
Comment (franckc) [inline contracts/contracts/vault/VaultAdmin.sol:246]: supernit: can add a break here to exit the loop and maybe save a tiny bit of gas.

**Review by franckc — APPROVED** [review]
Nice! One minor comment inline otherwise LGTM

**Pull request merged**